lib/API/JSON/API.php

```diff
@@ -36,6 +36,11 @@ class API {
   /** @var SettingsController */
   private $settings;
 
+  /** @var array */
+  private static $get_endpoints = [
+    'newsletterTemplates' => ['getAll'],
+  ];
```

Comment:
I'd recommend not doing this here in the parent API class. I'd suggest to keep it in endpoint class themselves:

1. Make it `protected static $get_methods = []`, leave it empty here.
2. In the inherited class itself make it `protected static $get_methods = ['getAll']`.
3. Use late static binding when accessing the array (`static::$get_methods` instead of `self::$get_methods`).